io-sim/src/Control/Monad/IOSim/Types.hs

```diff
@@ -141,6 +141,8 @@ data SimA s a where
   SetWallTime  ::  UTCTime -> SimA s b  -> SimA s b
   UnshareClock :: SimA s b -> SimA s b
 
+  StartTimeout :: DiffTime -> IOSim s a -> (Maybe a -> SimA s b) -> SimA s b
```

Comment:
Because `timeout :: forall a. DiffTime -> IOSim s a -> IOSim s (Maybe a)` but I guess I can make it be `SimA s a` not sure there will be any difference